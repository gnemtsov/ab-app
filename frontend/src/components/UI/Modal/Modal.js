import React, { Component } from 'react';

import classes from './Modal.css';
import Backdrop from '../Backdrop/Backdrop';

class Modal extends Component {

    shouldComponentUpdate ( nextProps, nextState ) {
        return nextProps.show !== this.props.show || nextProps.children !== this.props.children;
    }
    
    render () {
        return (
            <React.Fragment>
                <Backdrop show={this.props.show} clicked={this.props.modalClosed} />
                <div
                    className={classes.Modal}
                    style={{
                        transform: this.props.show ? 'translateY(0)' : 'translateY(-100vh)',
                        opacity: this.props.show ? '1' : '0'
                    }}>
                    {this.props.children}
                </div>
            </React.Fragment>
        )
    }
}

export default Modal;