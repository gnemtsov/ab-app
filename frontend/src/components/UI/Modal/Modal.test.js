import React from 'react';

import { configure, render, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Modal from './Modal';

configure({ adapter: new Adapter() });

describe('Modal', () => {
	it('Renders correctly when hidden', () => {
		const wrapper = shallow(<Modal/>);
		expect(wrapper).toMatchSnapshot();
	});
	it('Renders correctly when shown', () => {
		const wrapper = shallow(<Modal show />);
		expect(wrapper).toMatchSnapshot();
	});
	it('Renders correctly with children', () => {
		const wrapper = shallow(<Modal show>test</Modal>);
		expect(wrapper).toMatchSnapshot();
	});
	
	it('modalClosed', () => {
		const mockModalClosed = jest.fn(() => {});
		const wrapper = mount(<Modal show modalClosed={mockModalClosed}>test</Modal>);
		wrapper.find('backdrop').simulate('click');
		expect(mockModalClosed.mock.calls.length).toBe(1);
	});
})