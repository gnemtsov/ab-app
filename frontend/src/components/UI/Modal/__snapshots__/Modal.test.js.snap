// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Renders correctly when hidden 1`] = `
<Fragment>
  <backdrop />
  <div
    className="Modal"
    style={
      Object {
        "opacity": "0",
        "transform": "translateY(-100vh)",
      }
    }
  />
</Fragment>
`;

exports[`Modal Renders correctly when shown 1`] = `
<Fragment>
  <backdrop
    show={true}
  />
  <div
    className="Modal"
    style={
      Object {
        "opacity": "1",
        "transform": "translateY(0)",
      }
    }
  />
</Fragment>
`;

exports[`Modal Renders correctly with children 1`] = `
<Fragment>
  <backdrop
    show={true}
  />
  <div
    className="Modal"
    style={
      Object {
        "opacity": "1",
        "transform": "translateY(0)",
      }
    }
  >
    test
  </div>
</Fragment>
`;
